pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -168,54 +166,41 @@ func ConfigureTrustedCABundle(ctx context.Context, cli client.Client, log logr.L
 	return nil
 }
 
-// when DSCI TrustedCABundle.ManagementState is set to `Managed`.
-func AddCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsciv1.DSCInitialization) error {
-	namespaceList := &corev1.NamespaceList{}
-	if err := cli.List(ctx, namespaceList); err != nil {
-		return err
-	}
-
-	for i := range namespaceList.Items {
-		ns := &namespaceList.Items[i]
-		// check namespace status if not Active, then skip
-		if ns.Status.Phase != corev1.NamespaceActive {
-			continue
-		}
-
+// when DSCI TrustedCABundle.ManagementState is set to `Managed`, add new configmap into all active namespaces.
+func AddCABundleCMInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsciv1.DSCInitialization) error {
+	var multiErr *multierror.Error
+	err := cluster.ProcessAllNamespace(ctx, cli, func(ns *corev1.Namespace) error {
 		if ShouldInjectTrustedBundle(ns) {
-			if err := wait.PollUntilContextTimeout(ctx, time.Second*1, time.Second*10, false, func(ctx context.Context) (bool, error) {
+			if pollErr := wait.PollUntilContextTimeout(ctx, time.Second*1, time.Second*10, false, func(ctx context.Context) (bool, error) {
 				if cmErr := CreateOdhTrustedCABundleConfigMap(ctx, cli, ns.Name, dscInit.Spec.TrustedCABundle.CustomCABundle); cmErr != nil {
 					// Logging the error for debugging
 					fmt.Printf("error creating cert configmap in namespace %v: %v", ns.Name, cmErr)
 					return false, nil
 				}
 				return true, nil
-			}); err != nil {
-				return err
+			}); pollErr != nil {
+				multiErr = multierror.Append(multiErr, pollErr)
```

Comment:
It doesn't simplify much, but you don't have to check for nil. From the `multierror.Append` godoc:

> Any nil errors within errs will be ignored.